service/system_information.go

```diff
@@ -106,6 +106,21 @@ func (sh *Handler) CollectSystemInformation(ctx context.Context, connectInfo mul
 				continue
 			}
 
+			// Carefully handle disk with partitions.
+			// In case any of the partition on a disk is mounted, consider this disk to be used
+			// as the root disk of the system and therefore skip it including all of its partitions.
+			potentialRootDisk := false
+			for _, partition := range disk.Partitions {
+				if partition.Mounted {
+					potentialRootDisk = true
+					break
+				}
+			}
+
+			if potentialRootDisk {
+				continue
+			}
```

Comment:
This reads well but you could also consider labelling the outer loop and then having `continue outerLoop` instead of a `break` and a subsequent `if` block.

https://go.dev/ref/spec#Continue_statements